### ISSUE TITLE
`update` and `change` event should not be emitted if change is exogenous

### ISSUE THREAD

**Issue by yongjun21** [vueform/slider#4]
When value is updated through `v-model` (exogenous change) as opposed to user dragging the handle (endogenous change), there is no good reason to emit the `update` and `change` events since I can just watch on the value passed into `v-model`. This makes it hard to differentiate between exogenous and endogenous change

**Comment by oliv3** [comment]
I think I have the same issue. I use a Slider in a child component, the parent updates the slider's value through props. I need to send a command to a backend application when the slider changes (user sliding), but not when the parent updates it's value (through props+watch) ? Is that even possible ? Thanks !

**Comment by yongjun21** [comment]
Since this issue has been overlooked by the author, just putting my workaround here.

in your parent component, you can add a `willChange` variable that is initially set to false. Every time you trigger an exogenous change, set `willChange` to true. In your `onChange` listener, you can return early if `willChange` is true but always remember to reset `willChange` to false. I.E.

```vue
<template>
  <vue-slider v-model="value" @change="onChange"></vue-slider>
</template>

<script>
import VueSlider from '@vueform/slider'

export default {
  components: {VueSlider},
  data () {
    return {
      value: 0,
      willChange: false
    }
  },
  methods: {
    triggerChange (v) {
      this.willChange = true
      this.value = v
    },
    onChange () {
      if (!this.willChange) {
        // do something
      }
      this.willChange = false
    }
  }
}
</script>
```

**Comment by oliv3** [comment]
Yes ! Exactly what I thought of when I was running errands :)

**Comment by oliv3** [comment]
The only difference is that the Slider is inside a child component, but I'll give this a try.

**Comment by oliv3** [comment]
@yongjun21 it's working, but since I use Slider(s) in many child components, this is really hacky: i have to pass an object with `{ value, willChange }` to the prop and then have to watch with `deep: true` etc. And I have a dozen of Sliders in my application, so it will become quite ugly :( @adamberecz, any thoughts ? Thanks !

**Comment by oliv3** [comment]
@adamberecz Thanks, how can I test it ? Looks like it's still version 1.0.3 [here](https://www.npmjs.com/package/@vueform/slider).

**Comment by adamberecz** [comment]
Fixed in `1.0.4`
